python/cudf/utils/utils.py

```diff
@@ -146,9 +148,14 @@ def buffers_from_pyarrow(pa_arr, dtype=None):
         else:
             new_dtype = pa_arr.type.to_pandas_dtype()
 
-    padata = Buffer(
-        np.array(pa_arr.buffers()[1]).view(new_dtype)[
-            pa_arr.offset:pa_arr.offset + len(pa_arr)
-        ]
-    )
+    if buffers[1]:
+        padata = Buffer(
+            np.array(buffers[1]).view(new_dtype)[
+                pa_arr.offset:pa_arr.offset + len(pa_arr)
+            ]
+        )
+    else:
+        padata = Buffer(
+            np.empty(0, dtype=new_dtype)
+        )
```

Comment:
This looks like material that is piggybacking on this commit. Is there a test in place for this functionality? `if buffer[x]:` both look like `IndexErrors` to me.

Reply:
The reason for the `if buffers[1]` is that PyArrow can return a `None` object there. The only situation where there's only a single item in the `pa.buffers()` call is where we get a `NullArray` back from Arrow and we handle that separately for that very reason. There's some additional fixes for this in #920 and #939.